Exercise_4.py

```diff
@@ -1,38 +1,55 @@
-# Python program to insert element in binary tree  
-class newNode():  
-  
-    def __init__(self, data):  
-        self.key = data 
+# Python program to insert element in binary tree
+class newNode():
+
+    def __init__(self, data):
+        self.key = data
         self.left = None
         self.right = None
-          
-""" Inorder traversal of a binary tree"""
-def inorder(temp): 
-  
-    
-  
-  
-"""function to insert element in binary tree """
-def insert(temp,key): 
-  
-    
-  
-       
-# Driver code  
-if __name__ == '__main__': 
-    root = newNode(10)  
-    root.left = newNode(11)  
-    root.left.left = newNode(7)  
-    root.right = newNode(9)  
-    root.right.left = newNode(15)  
-    root.right.right = newNode(8)  
-  
-    print("Inorder traversal before insertion:", end = " ") 
-    inorder(root)  
-  
+
+def inorder(temp):
+    if temp:
+        inorder(temp.left)
+        print(temp.key)
+        inorder(temp.right)
+
+#source geeks for geeks for this function
+def insert(temp,key):
+    if not temp:
+        root = newNode(key)
+        return
+    L = []
+    L.append(temp)
+
+    while (len(L)):
+        temp = L[0]
+        L.pop(0)
```

Comment:
Causes all elements within the list to shift left, O(n) operation. Use a Queue or Dequeue instead.